### DISCUSSION TITLE
Is there any good ReactPHP based ORM package?

### DISCUSSION THREAD

**Discussion by nazmulpcc** [clue/framework-x#155] [Q&A] [answered]
Good Day everyone,
Does anyone know of any ORM packages that can be used with Framework X? As a big fan of Laravel, I use [illuminate/database](https://github.com/illuminate/database) or the Eloquent ORM in mid size projects. Is it feasible to modify it to play nice with ReactPHP?

**Comment by clue** [comment]
Good question and definitely an interesting feature! At the moment, I'm not aware of any async ORM implementation, but implementing one would be entirely feasible if there's enough demand. I'm sure we'll see an async DBAL and ORM implementation some time in the future! 🚀

Currently, there are no immediate plans to build this from my end (no demand at the moment and more important outstanding issues currently). If you need this for a project and you want to help work on this or sponsor this feature, feel free to reach out and I'm happy to take a look. :+1:

**Reply by nazmulpcc** [reply]
I do not need this for a project, but this & a queue processor is what's keeping me from using X in any real world project. I've also tinkered with some command line features like `php x shell` to open an interactive `psysh/psy` shell. I guess I will take a look at `doctrine/dbal` and see if I can make it work with ReactPHP.

**Comment by maciejfikus** [comment, marked as answer]
I may be a little late to the party, however there is quite nice package from DriftPHP folks. Works perfectly fine with X.
https://github.com/driftphp/reactphp-dbal
